documentation/docs/30-advanced/10-advanced-routing.md

```diff
@@ -123,27 +123,26 @@ src/routes/[...catchall]/+page.svelte
 
 ### Encoding
 
-Directory names are URI-decoded, meaning that (for example) a directory like `%40[username]` would match characters beginning with `@`:
-
-```js
-// @filename: ambient.d.ts
-declare global {
-	const assert: {
-		equal: (a: any, b: any) => boolean;
-	};
-}
-
-export {};
-
-// @filename: index.js
-// ---cut---
-assert.equal(
-	decodeURIComponent('%40[username]'),
-	'@[username]'
-);
-```
-
-To express a `%` character, use `%25`, otherwise the result will be malformed.
+Some characters can't be used on the filesystem — `/` on Linux and Mac, `\ / : * ? " < > |` on Windows. The `#` character has special meaning in URLs, and the `[ ] ( )` characters have special meaning to SvelteKit, so these also can't be used directly as part of your route.
+
+To use these characters in your routes, you can use HTML entities:
+
+- `\` — `&bsol;`
+- `/` — `&sol;`
```

Comment:
we cannot support `/`, `#`, or `?`. you must use a parameter for these. see the comment here https://github.com/sveltejs/kit/pull/7621/files#diff-a1a15b06db23e414ab4202232607550bcac6cfe2b0809777468e8dabb922b276R28

when we match we decode the URL with `decodeURI` and if you send an encoded version of these characters they will not be decoded, so you can't match them. they can only be sent via parameter